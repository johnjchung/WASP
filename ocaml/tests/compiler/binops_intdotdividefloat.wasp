Endpoint intDotDivideFloat(GET:) {
    float Test = 5 ./ 2.7;
}
