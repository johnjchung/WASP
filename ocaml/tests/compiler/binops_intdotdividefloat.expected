package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type intDotDivideFloat_struct struct {
	Test float64 `db:"test" json:"test"`
}

func main() {
	router := gin.Default()
	router.GET("/intDotDivideFloat", Get_intDotDivideFloat)
	router.Run(":"+os.Getenv("PORT"))
}

func intDotDivideFloat_struct_wrapper() intDotDivideFloat_struct {
	return intDotDivideFloat_struct{Test: float_conv(5) / float_conv(2.7)}
}

func Get_intDotDivideFloat(c *gin.Context) {
	intDotDivideFloat_struct := intDotDivideFloat_struct_wrapper()
	c.JSON(200, intDotDivideFloat_struct)
}
